g2p/convert.py

```diff
@@ -31,14 +36,29 @@ def run(self):
         with uopen(self.bliss_lexicon, "rt") as f:
             tree = ET.parse(f)
         with uopen(self.out_g2p_lexicon, "wt") as out:
-            for lemma in tree.findall(".//lemma"):
+            all_lemmas = tree.findall(".//lemma")
+            assert (
+                len(all_lemmas) > 0
+            ), "No lemma tag found in the lexicon file! Wrong format file?"
+
+            for lemma in all_lemmas:
                 if lemma.get("special") is not None:
                     continue
 
                 orth = lemma.find("orth").text.strip()
-                phon = lemma.find("phon").text.strip()
 
-                out.write("%s %s\n" % (orth, phon))
+                if self.write_first_phoneme_only:
+                    phon = lemma.find("phon").text.strip()
+                    out.write("%s %s\n" % (orth, phon))
+
+                else:
+                    phons = lemma.findall("phon")
+                    phon_single = []
```

Comment:
phon_single is used as a lookup datastructure here. In that case it's better to make it a set instead of a list as list has lookup time of O(n) vs. set's (amortized) O(1). In this case we do not expect many clashes or even entries it does not matter much, but I want to point out the general principle here.